### PR TITLE
[WIP - GH-42] Add Service Bus Subscription Count Client

### LINKED ISSUE #42: Add support for Service Bus Queue Topic Subscriptions
The primary External Metric provider uses Azure Monitor to gather metrics for the Service Bus Queues.  Service Bus only outputs total Metrics for a Queue Topic.  Using the GO [service bus library](https://github.com/Azure/azure-service-bus-go/blob/master/subscription.go#L56) we can query for the Subscription [value directly](https://github.com/Azure/azure-service-bus-go/blob/1912861e4166ab293c7ca0ac7d407d1b20a55a5e/subscription.go#L56).

### PR DESCRIPTION
**Pull request by marc-sensenich**
Allow for users of the metrics adapter to gather message counts from Azure Service Bus Topic Subscriptions to decide their scaling targets

Closes #42

### CODE REVIEW

File: pkg/azure/servicebus/subscription_count_client.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
+package servicebus
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/servicebus/mgmt/2017-04-01/servicebus"
```
Comment (marc-sensenich) [inline pkg/azure/servicebus/subscription_count_client.go]: I currently use the Azure SDK for Go as it allowed for the retrieval of each type of count within `CountDetails`. In the current case of `azure-service-bus-go` this number can be skewed if there are letters in the DLQ not being pulled

**Comment by marc-sensenich** [comment]
Based on #39, this is the structure I'd see for configuring the CRD
```yaml
apiVersion: azure.com/v1alpha2
kind: ExternalMetric
metadata:
  name: example-external-metric-service-bus-subscription
spec:
  type: servicebussubscription
  azure:
    resourceGroup: storage-rg
    namespace: sample-ns
    topic: sample-topic
    subscription: sample-subscription
  metric:
    # This would default to activeMessageCount, but could be updated to one of the counts from https://github.com/Azure/azure-sdk-for-go/blob/master/services/servicebus/mgmt/2017-04-01/servicebus/models.go#L1116
    field: activeMessageCount
```

File: pkg/azure/servicebus/subscription_count_client.go (hunk 1 above)
Comment (marc-sensenich) [reply pkg/azure/servicebus/subscription_count_client.go]: https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-dead-letter-queues

File: main.go [hunk 2]
```diff
@@ -67,10 +66,11 @@ func setupAzureProvider(cmd *basecmd.AdapterBase, metricsCache *metriccache.Metr
 	}
 
 	defaultSubscriptionID := getDefaultSubscriptionID()
+	serviceBusSubscriptionClient := servicebus.NewClient(defaultSubscriptionID)
 	monitorClient := monitor.NewClient(defaultSubscriptionID)
 	appinsightsClient := appinsights.NewClient()
 
-	azureProvider := azureprovider.NewAzureProvider(defaultSubscriptionID, mapper, dynamicClient, appinsightsClient, monitorClient, metricsCache)
+	azureProvider := azureprovider.NewAzureProvider(defaultSubscriptionID, mapper, dynamicClient, appinsightsClient, monitorClient, serviceBusSubscriptionClient, metricsCache)
```
[the remark this answers is not in the document]
Comment (marc-sensenich) [reply main.go]: Agreed, I'll move it towards the factory pattern

**Comment by marc-sensenich** [comment]
@jsturtevant currently didn't link up the providers just yet as I didn't want to get too far into the weeds without agreement from your end. I'll keep it moving forward based on the initial feedback

**Comment by jsturtevant** [comment]
Looks good to move forward on.  Let me know if there are any challenges or blockers.   Thanks again!

**Comment by jsturtevant** [comment]
Just checking in to see of there are any blockers I can help with.

**Comment by marc-sensenich** [comment]
No blockers currently. I got backlogged on professional and personal commitments and will be getting back to this PR.

**Comment by jsturtevant** [comment]
@marc-sensenich Thanks so much for this PR.  On Friday afternoon I was going to take the great start you have here and start to finish it up for the next release.  Let me know if you have any thoughts or objections.

**Comment by marc-sensenich** [comment]
@jsturtevant my apologies for leaving this one hanging. I have no objections from you getting this put together for the next release. I have some lingering changes I'll push up here and then you can take it or leave it.

**Comment by jsturtevant** [comment]
Thanks and no worries!  I appreciate the initial start to this.  I have a few spare cycles coming up and this is a frequently asked for feature.  So thank you for getting it started!

**Comment by jsturtevant** [comment]
I am closing this as I did not have permissions to push changes to @marc-sensenich branch.  Thanks for your initial implementation!

**Comment by marc-sensenich** [comment]
Thanks for taking this over @jsturtevant!

**Pull request closed without merging**